From b189bc9b1c4c6cafe76288f01a19d26a667dafc5 Mon Sep 17 00:00:00 2001
From: Allen Xu <allen.xu@freescale.com>
Date: Tue, 27 Mar 2012 18:12:13 +0800
Subject: [PATCH 292/505] ENGR00177747-2 add gpu unit test

1. use test-utils to test gpu module probe
2. simplify the Makefile

Signed-off-by: Allen Xu <allen.xu@freescale.com>
---
 test/gpu/Makefile | 3 ---
 test/gpu/gpu.sh   | 7 ++++---
 2 files changed, 4 insertions(+), 6 deletions(-)

diff --git a/test/gpu/Makefile b/test/gpu/Makefile
index 4a7ad57..7ce1008 100644
--- a/test/gpu/Makefile
+++ b/test/gpu/Makefile
@@ -3,9 +3,6 @@ INCLUDE_LIST:=IMX6Q
 
 ifeq ($(PLATFORM),$(findstring $(PLATFORM),$(INCLUDE_LIST)))
 OBJS = $(OBJDIR)/gpu.sh
-$(OBJS): $(SRCS)
-	mkdir -p $(OBJDIR)
-	cp gpu.sh $(OBJDIR)
 else
 OBJS =
 endif
diff --git a/test/gpu/gpu.sh b/test/gpu/gpu.sh
index 3f4110a..a5e2b95 100755
--- a/test/gpu/gpu.sh
+++ b/test/gpu/gpu.sh
@@ -6,9 +6,10 @@ source /unit_tests/test-utils.sh
 #
 pushd .
 #
-#insert gpu modules
+#run modprobe test
 #
-insmod galcore.ko
+gpu_mod_name=galcore.ko
+modprobe_test $gpu_mod_name
 #
 #run tests
 #
@@ -20,7 +21,7 @@ cd /opt/viv_samples/hal/ && ./tvui
 #
 #remove gpu modules
 #
-rmmod galcore.ko
+rmmod $gpu_mod_name
 #restore the directory
 #
 popd
-- 
1.8.0

